staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -351,6 +351,8 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 		}
 	}
 
+	klog.V(2).Infof("Caches populated for %v from %s", r.typeDescription, r.name)
```

Comment:
It seems to me that the `HasSynced` method is exposed on informers/controllers, not on reflectors. So the implementation can theoretically vary. Although, in practice, a synchronised reflector is indeed equivalent to a synchronised cache today.

Maybe it's better to identify the callers of the `WaitForCacheSync` method and check why they frequently use this method?

Reply:
Check why people call WaitForCacheSync?